vm/src/builtins/super.rs

```diff
@@ -110,10 +110,26 @@ impl PySuper {
     }
 
     #[pyproperty(magic)]
-    fn thisclass(&self) -> PyTypeRef {
+    fn __thisclass__(&self) -> PyTypeRef {
         self.typ.clone()
     }
 
+    #[pyproperty(magic)]
+    fn __self_class__(&self) -> Option<PyTypeRef> {
+        match &self.obj {
+            Some((_, obj_type)) => Some(obj_type.clone()),
+            _ => None,
+        }
```

Comment:
This code is also considerable.
```rust
self.obj.as_ref().map(|(_, obj_type)| obj_type.clone())
```
Same code can be used for `__self__`

Reply:
I prefer this form:
```suggestion
        Some(self.obj.as_ref()?.1.clone())
```